### PR TITLE
Artifact audit

### LINKED ISSUE #1333: Artifact audit
# What's needed

We're approaching release `0.9.0`, it's time to ensure all artifact names match the new approach ( `****-data` + `****-extensions`). We must also double check that artifacts mentioned in README + "Quick start" docs section also match with those names. Also review which ones are missing, if there are any.

## Audit results

The name on the left is the artifact name we're using in both README and Quick Start docs section. There's a ✅ in case both match with the name the artifact has in arrow codebase.

`arrow-core-data` :white_check_mark:
`arrow-core-extensions` :white_check_mark:
`arrow-syntax` :white_check_mark:
`arrow-typeclasses` :white_check_mark:
`arrow-extras` :boom: (should be `arrow-extras-data` ?)
`arrow-extras-extensions` :white_check_mark:
`arrow-meta` :white_check_mark:
`arrow-free`  -> :boom: it’s called `arrow-free-data`, should be updated in README and docs
`arrow-free-extensions` -> :white_check_mark:
`arrow-mtl` -> :white_check_mark:
`arrow-effects-data` -> :white_check_mark:
`arrow-effects-extensions` -> :white_check_mark:
`arrow-effects-io-extensions` -> :white_check_mark:
`arrow-effects-rx2-data` -> :white_check_mark:
`arrow-effects-rx2-extensions` -> :white_check_mark:
`arrow-effects-reactor-data` -> :white_check_mark:
`arrow-effects-reactor-extensions` -> :white_check_mark:
`arrow-effects-kotlinx-coroutines-data` -> :boom: where is the code?¿ remove leftovers in case we’re removing DeferredK and DeferredK extensions from Arrow
`arrow-effects-kotlinx-coroutines-extensions` -> :boom: where is the code?¿ remove leftovers in case we’re removing DeferredK and DeferredK extensions from Arrow
`arrow-optics` -> :white_check_mark:
`arrow-generic` -> :white_check_mark:
`arrow-recursion` -> :boom: it’s called `arrow-recursion-data`, we must rename it in both README and “Quick start” docs section.
`arrow-recursion-extensions` -> :white_check_mark:
`arrow-integration-retrofit-adapter` -> :white_check_mark:

## artifacts never mentioned in either README or docs Quick Start section

`arrow-query-language`
`arrow-kindedj`
`arrow-dagger`
`arrow-dagger-effects`
`arrow-dagger-effects-rx2`
`arrow-annotations`
`arrow-reflect`
`arrow-shadow`
`arrow-streams`
`arrow-validation`

Is it intended for all of them? should we mention any?

### PR DESCRIPTION
**Pull request by JorgeCastilloPrz**
Fixes #1333 

After some discussions there's an agreement on normalizing `***-data` and `***-extensions` for all required modules for `0.9.0` release, then we will iterate towards providing generic ones that provide bot things together for `0.9.1`. (as in exporting `core` which would include `core-data` and `core-extensions`, and same for other modules that contain the same duality).

* [x] renamed `arrow-extras` to `arrow-extras-data` and updated README and "Quick start" docs section accordingly.
* [x] renamed `arrow-free` to `arrow-free-data` in README and "Quick start" docs section since it was not matching. The artifact was correctly named.
* [x] removed deferredk leftovers and added a disclaimer in docs provided by @pakoito about why we've removed it from Arrow.
* [x] renamed `arrow-recursion` to `arrow-recursion-data` in both readme and "Quick start" docs section. The artifact was correctly named.
* [x] added `arrow-query-language` as optional to both readme and "Quick start" docs.

### CODE REVIEW

**Comment by JorgeCastilloPrz** [comment]
Is the docs page properly added? I think I'm missing linking to it from menus or something, can you point to what's missing in that sense please @pakoito @raulraja ?

File: modules/docs/arrow-docs/docs/docs/integrations/kotlinxcoroutines/README.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+---
```
Comment (pakoito) [inline modules/docs/arrow-docs/docs/docs/integrations/kotlinxcoroutines/README.md]: I don't think this is a new document, it should replace the existing one.

Comment (pakoito) [reply modules/docs/arrow-docs/docs/docs/integrations/kotlinxcoroutines/README.md]: Either that or github is trolling me.

Comment (JorgeCastilloPrz) [reply modules/docs/arrow-docs/docs/docs/integrations/kotlinxcoroutines/README.md]: It's actually the existing one but was blank

Comment (pakoito) [reply modules/docs/arrow-docs/docs/docs/integrations/kotlinxcoroutines/README.md]: https://raw.githubusercontent.com/arrow-kt/arrow/0.8.2/modules/docs/arrow-docs/docs/docs/integrations/kotlinxcoroutines/README.md

that's the raw file, match the header:

```
---
layout: docs
title: kotlinx.coroutines
permalink: /docs/integrations/kotlinxcoroutines/
---
```

and re-add the option in the menu, if possible:

https://github.com/arrow-kt/arrow/blob/0.8.2/modules/docs/arrow-docs/docs/_data/menu.yml#L416-L417

Comment (JorgeCastilloPrz) [reply modules/docs/arrow-docs/docs/docs/integrations/kotlinxcoroutines/README.md]: Ready, thx!

**Pull request merged**